atmo/jobs/models.py

```diff
@@ -99,13 +101,32 @@ def __repr__(self):
         return "<SparkJob {} with {} nodes>".format(self.identifier, self.size)
 
     @property
+    def has_never_run(self):
+        """
+        Whether the job has run before.
+        Looks at both the cluster status and our own record when
+        we asked it to run.
+        """
+        return (self.most_recent_status == self.DEFAULT_STATUS and
+                self.last_run_date is None)
+
+    @property
+    def has_finished(self):
+        """Whether the job's cluster is terminated or failed"""
+        return self.most_recent_status in self.FINAL_STATUS_LIST
+
+    @property
+    def is_runnable(self):
+        return self.has_never_run or self.has_finished
+
+    @property
     def is_expired(self):
-        """Tells whether the current job run has run out of time or not"""
-        if not self.current_run_jobflow_id or not self.last_run_date:
+        """Whether the current job run has run out of time"""
+        if not self.current_run_jobflow_id or self.last_run_date is None:
```

Comment:
You could use the new `has_never_run` property here and get rid of the check on `current_run_jobflow_id`. I think this condition at the moment only matches if the job has never run, because afaik we never reset those properties between jobs.